### PR TITLE
Add post about GPU CI in scikit-learn

### PR DESCRIPTION
**Pull request by betatim**
During the summit in 2024 I worked on getting the GPU CI for scikit-learn working. This is a post describing the setup
we ended up with. As well as a few other thoughts/things that were a bit unclear while setting up things.

There aren't any images yet, might attempt to generate one using "AI"

- [x] The main subject relates to at least one project affiliated to the Scientific Python Ecosystem.
- [x] I have the right to publish the content under BSD 3-Clause License for the code and Creative Common CC-BY-4.0 License for the text.
- [x] Images have been compressed using a tool like `pngquant`.

### CODE REVIEW

**Comment by betatim** [comment]
I tried to investigate the pre-commit failures, but when I try to run pre-commit locally it crashes with `OSError: [Errno 6] Device not configured`. Any ideas about how to fix that?


<details>
<summary>Details</summary>

### version information

```
pre-commit version: 3.7.1
git --version: git version 2.38.0
sys.version:
    3.12.4 | packaged by conda-forge | (main, Jun 17 2024, 10:13:44) [Clang 16.0.6 ]
sys.executable: /Users/thead/miniconda/envs/scientific-python-blog/bin/python3.12
os.name: posix
sys.platform: darwin
```

### error information

```
An unexpected error has occurred: OSError: [Errno 6] Device not configured
```

```
...
  File "/User/miniconda/envs/scientific-python-blog/lib/python3.12/site-packages/pre_commit/xargs.py", line 172, in run_cmd_partition
    return cmd_fn(
           ^^^^^^^
  File "/User/miniconda/envs/scientific-python-blog/lib/python3.12/site-packages/pre_commit/util.py", line 176, in cmd_output_p
    with open(os.devnull) as devnull, Pty() as pty:
  File "/User/miniconda/envs/scientific-python-blog/lib/python3.12/site-packages/pre_commit/util.py", line 133, in __enter__
    self.r, self.w = openpty()
                     ^^^^^^^^^
OSError: [Errno 6] Device not configured
```

</details>

**Comment by betatim** [comment]
pre-commit error fixed by rebooting

**Comment by betatim** [comment]
I've posted this to https://betatim.github.io/posts/github-action-with-gpu/, if we want to merge it here we need to add a `<link rel="canonical">` to it to refer to the original link so google doesn't get upset about duplicate content

File: content/posts/scikit-learn/gpu-ci/index.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
+---
+title: "Automated tests with GPUs for your project"
+date: 2024-07-19T15:24:17+01:00
+draft: false
+description: "Setting up CI with a GPU to test your code"
+tags: ["scikit-learn", "ci", "gpu", "cuda"]
+displayInList: true
+author: ["Tim Head <>timhead"]
```
Comment (stefanv) [inline content/posts/scikit-learn/gpu-ci/index.md:8]:
```suggestion
author: ["Tim Head <>timhead"]
```

Typo?

**Review by stefanv — APPROVED** [review]
Happy to see this merged, after verifying that author profile link works.

File: content/posts/scikit-learn/gpu-ci/index.md (hunk 1 above)
Comment (betatim) [reply content/posts/scikit-learn/gpu-ci/index.md:8]: Typo

**Comment by betatim** [comment]
Does anyone know how to add `<link rel="canonical" href="https://betatim.github.io/posts/github-action-with-gpu/" />` to the `<head>` of a post in hugo?

**Comment by stefanv** [comment]
See https://github.com/scientific-python/scientific-python-hugo-theme/pull/631

**Comment by betatim** [comment]
I think d338013 is the thing to do (I've not used Hugo before so a bit of a newbie). It doesn't seem to show up in the preview though :-/ Or is that because we need to update the version of the theme used?

**Comment by bsipocz** [comment]
> Or is that because we need to update the version of the theme used?

Correct, the theme is using the previous release. I suppose we could bump the submodule to be the current `main` of the theme? What do you think @stefanv and @jarrodmillman?

**Comment by stefanv** [comment]
@jarrodmillman will release a new theme soon, and once that's done we can merge this.

File: content/posts/scikit-learn/gpu-ci/index.md (hunk 1 above)
Comment (stefanv) [reply content/posts/scikit-learn/gpu-ci/index.md:8]: Err, still wrong... my suggestion was identical to the original :upside_down_face: ... should be `Tim Head <timhead>`.

**Comment by stefanv** [comment]
In it goes!

**Comment by betatim** [comment]
Thanks a lot!

**Pull request merged**